### PR TITLE
Created the /api end point and added a listen file

### PR DESCRIPTION
**Pull request by Karumu05**

### CODE REVIEW

File: __tests__/mainTestSuite.test.js [hunk 1]
```diff
@@ -31,13 +32,22 @@ describe('GET /api/topics', () => {
...
+    it('should respond with an accurate JSON describing other endpoints', () => {
         return request(app)
-        .get('/api/topics')
-        .expect(404)
+        .get('/api')
+        .expect(200)
         .then((response) => {
-            expect(response.body.msg).toBe('No topics exist')
+            const data = response.body.endpoints
+            expect(Object.keys(data)).toEqual(['GET /api', 'GET /api/topics', 'GET /api/articles'])
```
Comment (nc-poppy-m) [inline __tests__/mainTestSuite.test.js:44]: This is a brittle test and you will be required to change it as you develop your endpoints. How could you refactor this with matchers such as `.toHaveProperty` to make it more flexible?

File: app/models/endpoints.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
```
Comment (nc-poppy-m) [inline app/models/endpoints.json:1]: Is it me or do you have two endpoints files? One in models and one in root

**Review by nc-poppy-m — COMMENTED** [review]
Good work! have a look at your /api test (see below comments) , and also try to sort out the double endpoints files if there are indeed two of them.

File: app/models/endpoints.json (hunk 2 above)
Comment (Karumu05) [reply app/models/endpoints.json:1]: Yes sorry, i was playing around with trying to read the file through fs, but I could not get it to work so I copied it to see if I was getting the path wrong

Comment (Karumu05) [reply app/models/endpoints.json:1]: its been deleted

File: __tests__/mainTestSuite.test.js (hunk 1 above)
Comment (Karumu05) [reply __tests__/mainTestSuite.test.js:44]: I just encountered this issue and have resolved it

**Pull request merged**